src/storage-facade.ts

```diff
@@ -3,10 +3,37 @@ import * as FirebaseImp from "./firebase-db";
 import { IAnonymousPortalData, IPortalData } from "./portal-api";
 import { IExportableAnswerMetadata } from "./types";
 import {DexieStorage} from "./dexie-storage";
-
+import { refIdToAnswersQuestionId } from "./utilities/embeddable-utils";
 
 export interface IStorageInitializer { name: FirebaseImp.FirebaseAppName, preview: boolean }
 
+export interface IWrappedDBAnswer {
+  meta: IExportableAnswerMetadata;
+  interactiveState: any;
+}
+
+export type IIndexedDBAnswer = IExportableAnswerMetadata & { activity: string };
```

Comment:
Nice.  One of these days I'm going to sit down and figure out the difference between extending an `interface` and "intersection types" like this one (https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types).